stingray/simulator/events.py

```diff
@@ -174,3 +177,32 @@ def gen_lc_from_events(event_list, bin_time, start_time=None,
         times = times + bin_time / 2.
 
     return times, lc.astype(np.float)
+
+def assign_energies(N, spectrum):
+
+    '''
+    Assign energies to event lists.
+
+    Parameters
+    ----------
+    N: int
+        Length of event list
+    spectrum: 2-d array
+        Energies versus corresponding fluxes
+
+    Returns
+    -------
+    assigned_energies: array-like
+        Energies assigned to all events in event list
+    '''
+
+    assert spectrum.shape[0] != 2, "Spectrum must be a 2-d array!"
+
+    # Cast spectrum as numpy arrays and normalize by calculating cumulative density
+    fluxes = norm.cdf(np.array(spectrum[0]))
```

Comment:
Ok wrote custom code for that. 
Also, I have created a very basic test for now and all it does is ensure that the function is not breaking down. Is there a way to know what the expected energies should look like?